### PR TITLE
chore(sdk): add further zod support to SDK

### PR DESCRIPTION
**Pull request by nbayindirli**
### Description

* adds further zod support to SDK, namely for areas relating to `transactions`

### Drive-by changes

* No

### Related issues

- Fixes https://github.com/hyperlane-xyz/hyperlane-monorepo/issues/3740

### Backward compatibility

- Yes

### Testing

- None

### CODE REVIEW

File: typescript/sdk/src/index.ts
Comment (jmrossy) [inline typescript/sdk/src/index.ts:319]: I raised this question with @yorhodes and @ltyu too:
Is there value in having the schema and types be split across files this way, effectively doubling the file count for this kind of code? It seems to me that they're tightly coupled so there's no reason not to co-locate them.

File: typescript/sdk/src/providers/transactions/schemas.ts
Comment (jmrossy) [inline typescript/sdk/src/providers/transactions/schemas.ts:11]: I know this has been discussed before but can you clarify why we would want to allow non-primitive, lib-specific BigNumbers in our configs?

File: typescript/sdk/src/middleware/account/schemas.ts
Comment (jmrossy) [inline typescript/sdk/src/middleware/account/schemas.ts:18]: Maybe rename this? Having trouble understanding what this is for from the name. Or leave a comment

File: typescript/sdk/src/providers/transactions/transformer/TxTransformerTypes.ts
Comment (jmrossy) [inline typescript/sdk/src/providers/transactions/transformer/TxTransformerTypes.ts:2]: 👍

Comment (jmrossy) [reply typescript/sdk/src/providers/transactions/transformer/TxTransformerTypes.ts:2]: I'd be in favor of renaming the enum key value from `ICA` -> `InterchainAccount` for clarity

**Review by jmrossy — COMMENTED** [review]
This mostly lgtm! Just a few small questions please

File: typescript/sdk/src/index.ts
Comment (nbayindirli) [reply typescript/sdk/src/index.ts:319]: I am 100% on board with this. Faster to develop and fewer files. Perhaps we can rename all files to `zodTypes.ts`

Obviously– issue here is it would be a large refactor. The reason I did it this way is bc we were already doing it this way

@yorhodes wdyt

File: typescript/sdk/src/providers/transactions/schemas.ts
Comment (nbayindirli) [reply typescript/sdk/src/providers/transactions/schemas.ts:11]: afaik we do not want this.. ideally we would have this: https://github.com/hyperlane-xyz/hyperlane-monorepo/issues/3771

line 11 exists here just to unblock me with getting these zod schemas completed

File: typescript/sdk/src/middleware/account/schemas.ts
Comment (nbayindirli) [reply typescript/sdk/src/middleware/account/schemas.ts:19]: Added comment `/* For InterchainAccount::getCallRemote() */`

File: typescript/sdk/src/providers/transactions/transformer/TxTransformerTypes.ts
Comment (nbayindirli) [reply typescript/sdk/src/providers/transactions/transformer/TxTransformerTypes.ts:2]: Keen– will make update now

File: typescript/sdk/src/providers/transactions/schemas.ts
Comment (yorhodes) [reply typescript/sdk/src/providers/transactions/schemas.ts:11]: the workaround should be z.string() for any "bignumberish" types before 3771

File: typescript/sdk/src/providers/transactions/transformer/TxTransformerTypes.ts
Comment (cmcewen) [inline typescript/sdk/src/providers/transactions/transformer/TxTransformerTypes.ts:2]: is this exported outside of the SDK? would this be a breaking change?

File: typescript/sdk/src/index.ts
Comment (cmcewen) [reply typescript/sdk/src/index.ts:319]: rossy your comment makes sense but in the context of this PR i do think it's good for noah to land it following existing convention, and then we can maybe just do a chore refactor of all of it at once

Comment (jmrossy) [reply typescript/sdk/src/index.ts:319]: @cmcewen Yep agreed, just meant to propose the change

File: typescript/sdk/src/providers/transactions/transformer/TxTransformerTypes.ts
Comment (nbayindirli) [reply typescript/sdk/src/providers/transactions/transformer/TxTransformerTypes.ts:2]: Yes, and yes in isolation it is breaking. It's currently only consumed by my submit changes on CLI-side, which is not exposed via a command yet. This PR makes that change CLI-side [here](https://github.com/hyperlane-xyz/hyperlane-monorepo/blob/27be8100d7db710277cfb166f3e1219f146907e0/typescript/cli/src/submit/submit.ts#L78)

File: typescript/sdk/src/index.ts
Comment (nbayindirli) [reply typescript/sdk/src/index.ts:319]: Issue to capture: https://github.com/hyperlane-xyz/hyperlane-monorepo/issues/3844

File: typescript/sdk/src/providers/transactions/schemas.ts
Comment (yorhodes) [inline typescript/sdk/src/providers/transactions/schemas.ts:18]: why have 2 different schemas which share 3 of same fields?
also `to` is `ZHash` on `CallData` but `z.string` on `PopulatedTransaction`

File: typescript/sdk/src/providers/transactions/submitter/ethersV5/types.ts
Comment (yorhodes) [inline typescript/sdk/src/providers/transactions/submitter/ethersV5/types.ts:1]: these names are a little longer than I personally prefer but more of a taste thing

File: typescript/sdk/src/providers/transactions/transformer/ethersV5/schemas.ts
Comment (yorhodes) [inline typescript/sdk/src/providers/transactions/transformer/ethersV5/schemas.ts:7]: any intended use for hookMetadata atm?

**Review by yorhodes — COMMENTED** [review]
would be great to have some super basic unit tests for your schemas

File: typescript/sdk/src/providers/transactions/transformer/ethersV5/schemas.ts
Comment (nbayindirli) [reply typescript/sdk/src/providers/transactions/transformer/ethersV5/schemas.ts:7]: None, just conforming to getCallRemote since that's out of scope for this change

File: typescript/sdk/src/providers/transactions/schemas.ts
Comment (nbayindirli) [reply typescript/sdk/src/providers/transactions/schemas.ts]: > why have 2 different schemas which share 3 of same fields?

to emulate the current state of the repository. we can eventually refactor everything to just use a single in-house type. captured here: https://github.com/hyperlane-xyz/hyperlane-monorepo/issues/3845

> also to is ZHash on CallData but z.string on PopulatedTransaction

did not see this– updated PopulatedTransaction to use ZHash

**Review by paulbalaji — APPROVED** [review]

**Review by jmrossy — APPROVED** [review]

**Comment by jmrossy** [comment]
Please add a changeset before merging

File: typescript/sdk/src/providers/transactions/submitter/builder/schemas.ts
Comment (ltyu) [inline typescript/sdk/src/providers/transactions/submitter/builder/schemas.ts:7]: should this also use `ZChainName`?

**Pull request merged**
